python/src/worker/tasks/background_expressed_genes.py

```diff
@@ -50,15 +50,18 @@ def compute(self):
         request = self._format_request()
 
         # send request to r worker
-        r = requests.post(
+        response = requests.post(
             f"{config.R_WORKER_URL}/v0/getBackgroundExpressedGenes",
             headers={"content-type": "application/json"},
             data=json.dumps(request),
         )
 
-        # raise an exception if an HTTPError if one occurred because otherwise r.json()
+        # raise an exception if an HTTPError if one occurred because otherwise response.json()
         #  will fail
-        r.raise_for_status()
-        r = r.json()
+        response.raise_for_status()
+        result = response.json()
+        self.set_error(result)
+        if self.error:
+            return self._format_result(None)
```

Comment:
I don't see the benefit of adding the error to the class. Why not just:
```
result = response.json()
error = response.get("error", False)
return self._format_result(result, error)
```
and have format result handle removing the result if there are errors